src/mrsimulator/post_simulation.py

```diff
@@ -1,98 +1,206 @@
 
-def Lorentzian(csdm, axis, arg, **kwargs):
-    """Lorentzian apodization function.
+        data: CSDM object
+        dep_var: int. The index of the dependent variable to apply operation to
+        """
+        data_copy = data.copy()
+        return data_copy.fft(axis=self.dimension)
 
-    Args:
-        Self: simulation object
-        arg: list with one entry. The full-width-half-max in Hz of the Lorentzian function
 
-    Returns:
-        A Numpy array
+class FFT(abstractOperation):
     """
-    inv_x = csdm.dimensions[axis].reciprocal_coordinates().to("s").value
-    return np.exp(-arg[0] * np.pi * np.abs(inv_x))
+    abstractOperation-based class for applying a
+    Fourier transform to a dependent variable of simulation data
 
+    dimension: int. Data dimension to apply the function along
 
-def Gaussian(csdm, axis, arg, **kwargs):
-    """Gaussian apodization function.
+    """
 
-    Args:
-        Self: simulation object
-        sigma: list with one entry. standard deviation in Hz of the Gaussian function
+    dimension: int = 0
 
-    Returns:
-        A Numpy array
+    def operate(self, data, dep_var):
+        """
+        Applies the operation for which the class is named for.
+
+        data: CSDM object
+        dep_var: int. The index of the dependent variable to apply operation to
+        """
+        data_copy = data.copy()
+        return data_copy.fft(axis=self.dimension)
+
+
+class OperationList(BaseModel):
+    """
+    Container class to hold a list of operations to be applied to a given
+    dependent variable in the data
+
+    dependent_variable: The dependent variable of the data to apply operations to
+    operations: List of abstractOperation based operations to sequentially apply to data.
     """
-    inv_x = csdm.dimensions[axis].reciprocal_coordinates().to("s").value
-    return np.exp(-((inv_x * arg[0] * np.pi) ** 2) * 2)
 
+    dependent_variable: int = 0
+    operations: List[abstractOperation]
 
-class PostSimulator(Parseable):
-    """scale: scaling factor
+
+class SignalProcessor(BaseModel):
+    """
+    Signal processing class to apply lists of various operations to individual dependent variables
+    of the data.
+
+    data: CSDM object. From simulation
+    operations: List of operation lists
     """
 
-    scale: Optional[float] = 1.0
-    apodization: List[Apodization] = []
-    truncation_artifact: Optional[bool] = False
-    dead_time: Optional[float] = 0.0
+    data: Union[cp.CSDM, np.ndarray] = None
+    operations: List[OperationList]
 
     class Config:
         validate_assignment = True
         arbitrary_types_allowed = True
+
+    def apply_operations(self, **kwargs):
+        op_list = self.operations
+
+        for item in op_list:
+            dep_var = item.dependent_variable
+            for filters in item.operations:
+                self.data = filters.operate(self.data, dep_var=dep_var)
+
+        return self.data
```

Comment:
Instead of copying the data in every `def operate` function from all operation classes, copy the data one-time in the `apply_operations` function, and perform the operation on the copy.

```
          copy_data = self.data.copy()

          for item in op_list:
              dep_var = item.dependent_variable
              for filters in item.operations:
                  copy_data = filters.operate(copy_data, dep_var=dep_var)
 
          return copy_data
```